targets/mbed/source/jerry_extapi.cpp

```diff
@@ -86,41 +87,37 @@ static bool
 register_native_function (const char* name,
                           jerry_external_handler_t handler)
 {
-  jerry_api_object_t *global_obj_p;
-  jerry_api_object_t *reg_func_p;
-  jerry_api_value_t reg_value;
-  bool bok;
+  jerry_value_t global_object_val = jerry_get_global_object ();
+  jerry_value_t reg_function = jerry_create_external_function (handler);
+  jerry_value_t set_result;
+  bool is_ok = true; 
 
-  global_obj_p = jerry_api_get_global ();
-  reg_func_p = jerry_api_create_external_function (handler);
-
-  if (!(reg_func_p != NULL
-      && jerry_api_is_function (reg_func_p)
-      && jerry_api_is_constructor (reg_func_p)))
+  if (!(jerry_value_is_function (reg_function)
+        && jerry_value_is_constructor (reg_function)))
```

Comment:
I think the error flag should be checked. The returned value can be an object, which can also be a constructor.